utensor_cgen/ir/misc/graph_viz.py

```diff
@@ -65,3 +70,82 @@ def viz_graph(ugraph, out_fname=None, view=False, cleanup=True, colored_nodes=No
+      logger.info('No tensor allocation plan to visualize: %s', ugraph.name)
+      return plt.Figure()
+    alloc_plan = ugraph.attributes[TensorAllocationPlanner.KWARGS_NAMESCOPE]
+    topo_tensors = sorted(
+      [tensor_name for tensor_name in alloc_plan.plan],
+      key=lambda tensor_name: alloc_plan.plan[tensor_name].time_slot_start
+    )
+    return cls._draw_figs(topo_tensors, alloc_plan, cmap, figsize, fontsize, lw, split_on_large_graph, num_tensors_per_split)
+  
+  @classmethod
+  def _draw_figs(cls, topo_tensors, alloc_plan, cmap, figsize, fontsize, lw, split_on_large_graph, num_tensors_per_split):
+    xmins = [alloc_plan.plan[tensor_name].offset_start for tensor_name in topo_tensors]
+    xmaxs = [alloc_plan.plan[tensor_name].offset_end for tensor_name in topo_tensors]
+    colors = [cmap(random()) for _ in alloc_plan.plan]
+    labels = topo_tensors[:]
+    sizes = [alloc_plan.plan[tensor_name].size for tensor_name in topo_tensors]
+    if split_on_large_graph:
+      xmin_chunks = [xmins[i:i+num_tensors_per_split] for i in range(0, len(xmins), num_tensors_per_split)]
+      xmax_chunks = [xmaxs[i:i+num_tensors_per_split] for i in range(0, len(xmaxs), num_tensors_per_split)]
+      color_chunks = [colors[i:i+num_tensors_per_split] for i in range(0, len(colors), num_tensors_per_split)]
+      label_chunks = [labels[i:i+num_tensors_per_split] for i in range(0, len(labels), num_tensors_per_split)]
+      size_chunks = [sizes[i:i+num_tensors_per_split] for i in range(0, len(sizes), num_tensors_per_split)]
+    else:
+      xmin_chunks = [xmins]
+      xmax_chunks = [xmaxs]
+      color_chunks = [colors]
+      label_chunks = [labels]
+      size_chunks = [sizes]
+    figs = []
+    for i, (xmins, xmaxs, colors, labels, sizes) in enumerate(zip(xmin_chunks, xmax_chunks, color_chunks, label_chunks, size_chunks)):
+      fig, _ = plt.subplots(1, 1)
+      ys = [len(xmins)-i for i in range(len(xmins))]
+      for y, xmin, xmax, color, size in zip(ys, xmins, xmaxs, colors, sizes):
+        plt.hlines(y, xmin, xmax, lw=lw, colors=color)
+        plt.text(xmax+lw*10, y-0.01*lw, '{} bytes'.format(size), fontdict={'fontsize': fontsize})
+      plt.xlabel('Offset (bytes)', fontdict={'fontsize': fontsize})
+      plt.yticks(ys, labels, fontsize=fontsize)
+      plt.xticks(fontsize=fontsize)
+      plt.ylabel(
+        'Tensor Names (Topological Ordered, Top to Bottom)',
+        fontdict={'fontsize':fontsize}
+      )
+      if i:
+        title = 'Optimal Tensor Allocation: {} bytes in total (Cont.)'.format(alloc_plan.total_size)
+      else:
+        title = 'Optimal Tensor Allocation: {} bytes in total'.format(alloc_plan.total_size)
+      plt.title(
+        title,
+        fontdict={'fontsize': fontsize}
+      )
+      if figsize is None:
+        figsize = (num_tensors_per_split, num_tensors_per_split / 2)
+      fig.set_size_inches(*figsize)
+      fig.tight_layout()
+      figs.append(fig)
```

Comment:
Why not use subfigs?

Reply:
You are right.
I just didn't think about it. XD

Reply:
lolol I have a massive amount of experience with matlab so this is in my blood XD

Reply:
Ok, I see.
BTW, I really hate matlab plotting api.
I swear to gods I'll replace `matplotlib` with `ggplot2` once I get a chance.

Reply:
both work, I have more experience with matlab style plots but I have no issues with switching